holidays/countries/norfolk_island.py

```diff
@@ -0,0 +1,137 @@
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    MON_TO_NEXT_TUE,
+    SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class NorfolkIsland(
+    ObservedHolidayBase,
+    ChristianHolidays,
+    InternationalHolidays,
+    StaticHolidays,
+):
+    """Norfolk Island holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Norfolk_Island>
+        * <https://web.archive.org/web/20250409233237/https://www.infrastructure.gov.au/territories-regions-cities/territories/norfolk-island/travel-information>
+        * [2021](https://web.archive.org/web/20250710234146/http://www.norfolkislander.com/images/2020_07_31_Gazette_No._35.pdf)
+        * [2022](https://web.archive.org/web/20250328071352/https://www.nirc.gov.au/files/assets/public/v/1/your-council/documents/nirc-gazettes/2021_07_09_gazette_no_29.pdf)
+        * [2025](https://web.archive.org/web/20250711000525/https://www.nirc.gov.au/files/assets/public/v/1/your-council/documents/nirc-gazettes/2024_07_05_gazette_no_25.pdf)
+    """
+
+    country = "NF"
+    default_language = "en_NF"
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (observed, estimated)")
+    supported_languages = ("en_NF", "en_US")
+    start_year = 2016
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, NorfolkIslandStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Australia Day.
+        self._add_observed(self._add_holiday_jan_26(tr("Australia Day")))
+
+        # Foundation Day.
+        self._add_observed(self._add_holiday_mar_6(tr("Foundation Day")), rule=SUN_TO_NEXT_MON)
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        # ANZAC Day.
+        self._add_observed(self._add_anzac_day(tr("ANZAC Day")))
+
+        # Bounty Day.
+        self._add_observed(self._add_holiday_jun_8(tr("Bounty Day")), rule=SAT_SUN_TO_NEXT_MON)
+
+        sovereign_birthday_dates = {
+            2019: (JUN, 17),
+            2024: (JUN, 17),
+        }
+        # Sovereign's Birthday.
+        if self._year >= 2023:
+            name = tr("King's Birthday")
+        else:
+            name = tr("Queen's Birthday")
+
+        if dt := sovereign_birthday_dates.get(self._year):
+            self._add_holiday(name, dt)
+        else:
+            self._add_observed(
+                self._add_holiday_2_days_past_2nd_sat_of_jun(name),
+            )
```

Comment:
```suggestion
            self._add_holiday_2_days_past_2nd_sat_of_jun(name)
```
And we need to consider this:

>if that Monday is observed as Bounty Day, the next Monday

Reply:
> if that Monday is observed as Bounty Day, the next Monday

is there any example for this situation other than children's day in taiwan?

Reply:
National Day in Hong Kong, Mother's Day in Madagaskar, Näfelser Fahrt in Switzerland... ;)